hypervisor/src/hypervisor.rs

```diff
@@ -110,12 +110,16 @@ pub trait Hypervisor: Send + Sync {
     /// Create a Vm using the underlying hypervisor
     /// Return a hypervisor-agnostic Vm trait object
     ///
-    fn create_vm(&self) -> Result<Arc<dyn Vm>>;
+    fn create_vm(&self, #[cfg(feature = "sev_snp")] _mem_size: u64) -> Result<Arc<dyn Vm>>;
     ///
     /// Create a Vm of a specific type using the underlying hypervisor
     /// Return a hypervisor-agnostic Vm trait object
     ///
-    fn create_vm_with_type(&self, _vm_type: u64) -> Result<Arc<dyn Vm>> {
+    fn create_vm_with_type(
+        &self,
+        _vm_type: u64,
+        #[cfg(feature = "sev_snp")] _mem_size: u64,
+    ) -> Result<Arc<dyn Vm>> {
```

Comment:
I would rather you add a specific new entry point here rather than overloading the existing ones.

Reply:
Please take a look now.